core/src/test/scala/unit/kafka/utils/TestUtils.scala

```
@@ -1640,6 +1800,37 @@ object TestUtils extends Logging {
     }, s"Timed out waiting for brokerId $brokerId to come online")
   }
 
+  def getReplicaAssignmentForTopics[B <: KafkaBroker](
+      topicNames: Seq[String],
+      brokers: Seq[B],
+      adminConfig: Properties = new Properties): Map[TopicPartition, Seq[Int]] = {
+    val adminClient = createAdminClient(brokers, adminConfig)
+    val results = new mutable.HashMap[TopicPartition, Seq[Int]]
+    try {
+      adminClient.describeTopics(topicNames.toList.asJava).topicNameValues().forEach {
+        case (topicName, future) =>
+          try {
+            val description = future.get()
+            description.partitions().forEach {
+              case partition =>
+                val topicPartition = new TopicPartition(topicName, partition.partition())
+                results.put(topicPartition, partition.replicas().asScala.map(_.id))
+            }
+          } catch {
+            case e: ExecutionException => if (e.getCause != null &&
+              e.getCause.isInstanceOf[UnknownTopicOrPartitionException]) {
+              // ignore
```

Wondering if it would be better to let this propagate instead of returning an empty result. At least maybe we can mention the behavior in a doc comment.

---

I believe this is consistent with the behavior we have in the ZK case (which is not in TestUtils, but in KafkaZkClient or somewhere, as I recall)...

I will add a JavaDoc comment.